src/Dropdown/Dropdown.tsx

```diff
@@ -7,32 +7,26 @@ import { Popper, PopperProps, Placement } from './Popper'
 
 export const DropdownPlacement = Placement
 
-export interface DropdownProps<T> extends Pick<PopperProps, 'open' | 'placement' | 'onOutsideClick' | 'flip'> {
+export interface DropdownProps extends Pick<PopperProps, 'open' | 'placement' | 'onOutsideClick' | 'flip'> {
   children: React.ReactNode
+  trigger: React.ReactNode
   useTriggerWidth?: boolean
-  TriggerComponent: React.ComponentType<T & { ref: React.Ref<any>, onClick?: () => void }>
-  triggerComponentProps?: T
-  onClick?: () => void
 }
 
-export function Dropdown <T extends {}> ({
-  TriggerComponent,
-  triggerComponentProps = {} as T,
+export function Dropdown ({
+  trigger,
   open,
   flip,
   children,
   useTriggerWidth = false,
   placement,
   onOutsideClick,
-  onClick,
   ...blockProps
-}: DropdownProps<T> & BlockProps) {
+}: DropdownProps & BlockProps) {
   return (
     <Popper
       renderTrigger={({ ref }) => (
-        <div ref={ref}>
-          <TriggerComponent onClick={onClick} {...triggerComponentProps} />
-        </div>
+        <div ref={ref}>{trigger}</div>
```

Comment:
Would we need a way to style this <div>? Because now the trigger will always be rendered as a block element in HTML, and maybe we need it to be e.g. inline or flex, but there seems no way we can have control over this now?

Reply:
Could do it following way, so you do not have to render a div:
```
 renderTrigger={({ ref }) => cloneElement(trigger, { ref })}
```